xtable-core/src/test/java/org/apache/xtable/ITConversionController.java

```diff
@@ -797,15 +883,45 @@ private void checkDatasetEquivalence(
             // if count is not known ahead of time, ensure datasets are non-empty
             assertFalse(dataset1Rows.isEmpty());
           }
+          // Process UUID for the diff in Iceberg and Delta to make test smooth
+          // Iceberg supports UUIDs directly, while Delta represents them as binary
+          List<String> processedDataset1Rows = processUUID(dataset1Rows);
+          List<String> processedDataset2Rows = processUUID(dataset2Rows);
           assertEquals(
-              dataset1Rows,
-              dataset2Rows,
+              processedDataset1Rows,
+              processedDataset2Rows,
               String.format(
                   "Datasets are not equivalent when reading from Spark. Source: %s, Target: %s",
                   sourceFormat, format));
         });
   }
 
+  // converting Base64-encoded UUID fields from Delta to their UUID string representations for
+  // testing
+  private List<String> processUUID(List<String> rows) {
+    Gson gson = new Gson();
+    JsonParser parser = new JsonParser();
```

Comment:
We use jackson in other places for json, can we use that here to avoid requiring another library in case our dependencies change in the future?

Reply:
> We use jackson in other places for json, can we use that here to avoid requiring another library in case our dependencies change in the future?

Sure, thanks for the context! Already replaced that in latest commit.